src/main/scala/Clock.scala

```diff
@@ -4,38 +4,44 @@ import scala.concurrent.duration._
 
 import java.text.DecimalFormat
 
-// All durations are expressed in seconds
+// All unspecified durations are expressed in seconds
 sealed trait Clock {
+  @inline implicit def CentisWrapper(c: Int) = new Centis(c)
+  @inline implicit def CentisWrapper(c: Long) = Centis(c)
+
   val config: Clock.Config
   val color: Color
-  val whiteTime: Float
-  val blackTime: Float
-  val timerOption: Option[Double]
+  val whiteTime: Centis
+  val blackTime: Centis
+  val whiteBerserk: Boolean
+  val blackBerserk: Boolean
+  val timerOption: Option[Timestamp]
 
+  def limitSeconds = config.limitSeconds
   def limit = config.limit
-  def increment = config.increment
 
-  def time(c: Color) = c.fold(whiteTime, blackTime)
+  def incrementSeconds = config.incrementSeconds
+  def increment = config.increment
 
-  def outoftime(c: Color) = remainingTime(c) == 0
+  def time(c: Color): Centis = c.fold(whiteTime, blackTime)
 
-  def outoftimeWithGrace(c: Color, graceMillis: Int) =
-    millisSinceFlag(c).exists(graceMillis.min(Clock.maxGraceMillis).<)
+  def outoftime(c: Color) = remainingTime(c).value == 0
 
-  def remainingTime(c: Color) = math.max(0, limit - elapsedTime(c))
+  def outoftimeWithGrace(c: Color, grace: Centis) =
+    timeSinceFlag(c).exists((grace atMost Clock.maxGrace).<)
 
-  def remainingDuration(c: Color): FiniteDuration =
-    (remainingTime(c) * 1000).toLong.millis
+  def remainingTime(c: Color): Centis = limit - elapsedTime(c) nonNeg
```

Comment:
this looks ambiguous. are you sure `nonNeg` applies to the full expression, after the subtraction is done? a couple parens would make it clearer.